src/db/util/async.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * Create an async function that can be called multiple times with different
+ * arguments, and result in a single invocation of an async function that
+ * serves all the given calls at once.
+ *
+ * This function takes advantage of the NodeJS single-threaded concurrency model
+ * to group multiple requests together using `setImmediate`.
+ */
+export const createGroupableAsyncFunction = <
```

Comment:
this function can apply more generally than jsut `db` related actions, so should probably live in a module in `src/util/async` instead.